tfhe/src/core_crypto/algorithms/lwe_encryption.rs

```diff
@@ -377,8 +472,8 @@ where
     *output_body.data = encoded.0;
 
     let ciphertext_modulus = output_body.ciphertext_modulus();
-    assert!(ciphertext_modulus.is_compatible_with_native_modulus());
-    if !ciphertext_modulus.is_native_modulus() {
+    // Manage the non native power of 2 encoding
+    if ciphertext_modulus.kind() == CiphertextModulusKind::NonNativePowerOfTwo {
         *output_body.data = (*output_body.data)
             .wrapping_mul(ciphertext_modulus.get_power_of_two_scaling_to_native_torus());
```

Comment:
Would be cleaner to have 
```
 *output_body.data = match kind{
    nnp2 => encoded * scaling,
    _ => encoded
}
```

Maybe better to call `trivially_encrypt_lwe_ciphertext` instead, even if it fills `0`s twice